3.resistance-signature/scripts/nbconverted/11.1.last-batch-visualize-otherclone-signature-application_validation.r

```diff
@@ -0,0 +1,190 @@
+    #tidyr::spread(`AUROC\nShuffled:`, roc_auc, sep="") %>%
+    #dplyr::arrange(desc(Split))
+
+#roc_df$Split <- dplyr::recode(roc_df$Split, !!!legend_labels)
+#roc_df
+
+table_theme <- gridExtra::ttheme_default(
+    core = list(fg_params=list(cex = 0.3)),
+    colhead = list(fg_params=list(cex = 0.3)),
+    padding = unit(c(1.5, 1.5), "mm")
+)
+
+table_gg <- gridExtra::tableGrob(roc_df,
+                                 theme = table_theme,
+                                 rows = NULL)
+
+threshold_points_df <- roc_curve_df %>%
+    dplyr::filter(shuffled == "False") %>%
+    #dplyr::group_by(model_split) %>%
+    dplyr::filter(abs(threshold) == min(abs(threshold))) %>%
+    dplyr::ungroup()
+
+#legend_colors <- c("#D41159")
+#legend_labels <- c("WT" = "WT", "BZ" = "BZ")
+
+roc_gg <- (
+    ggplot(roc_curve_df, aes(x = fpr, y = tpr))
+    + geom_line(aes(linetype = shuffled), size=0.2)
+    + geom_point(data = threshold_points_df, size = 1)
+    + geom_abline(intercept=0, slope=1, size=0.2, linetype="dashed", color="black")
+    + scale_color_manual(name="Dataset", labels=legend_labels, values=legend_colors)
+    + scale_linetype_manual(name="Shuffled", labels=linetype_labels, values=linetype_default)
+    + coord_fixed()
+    + xlab("False positive rate")
+    + ylab("True positive rate")
+    + annotation_custom(table_gg, ymin = -0.82, xmin = 0.42)
+    + custom_theme
+    + theme(legend.position = "bottom")
+    + guides(col = guide_legend(nrow = 2, byrow = TRUE), linetype = guide_legend(nrow = 2, byrow = TRUE))
+)
+
+min_val <- unique(signature_df$min_permuted_value)
+max_val <- unique(signature_df$max_permuted_value)
+
+box_plot_gg <- (
+    ggplot(signature_df, aes(x = Metadata_clone_number, y = TotalScore))
+    + geom_boxplot(aes(fill = Metadata_clone_type), lwd = 0.1, alpha = 0.9, outlier.alpha = 0)
+    + geom_jitter(
+        position = position_jitterdodge(jitter.width = 0.4, dodge.width = 0.75),
+        aes(fill = Metadata_clone_type),
+        stroke = 0.2,
+        alpha = 0.6,
+        shape = 21,
+        lwd = 0.5
+    )
+    #+ facet_wrap(
+       # "~Metadata_dataset",
+       # ncol = 1,
+       # scales = "free_y")
+    
+    + xlab("")
+    + ylab("Signature score\n(singscore)")
+    + scale_fill_manual("Resistance status", values = resistance_status_colors, labels = resistance_status_labels)
+    + scale_y_continuous(breaks = c(-0.6, -0.4, -0.2, 0, 0.2, 0.4, 0.6), minor_breaks = seq(-10, 10, 0.1))
+    + geom_hline(yintercept = 0, linetype = "dashed", size = 0.2, color = "black")
+    + coord_flip()
+    + annotate(
+        "rect",
+         ymin = min_val,
+         ymax = max_val,
+         xmin = 0.3,
+         xmax = 11,
+         alpha = 0.2,
+         color = "red",
+         linetype = "dashed",
+         fill = "grey",
+         size = 0.1
+    )
+    + custom_theme
+    + guides(fill = guide_legend(reverse = TRUE))
+)
+
+output_fig_file <- file.path("figures", "newbatch_validation_roc.png")
```

Comment:
I recommend trying to keep the naming convention consistent between the input dataset, notebook file name, and any outputs.

This means not naming the png files `newbatch` but keeping it consistent with `last-batch` (which, btw, I am not a fan of the name, since we don't _actually_ know if it is the last batch or not 😉 )